InvenTree/build/models.py

```diff
@@ -916,6 +916,10 @@ def complete_build_output(self, output, user, **kwargs):
         # List the allocated BuildItem objects for the given output
         allocated_items = output.items_to_install.all()
 
+        if (common.settings.prevent_build_output_complete_on_incompleted_tests() and output.hasRequiredTests() and not output.passedAllRequiredTests()):
+            raise ValidationError(
+                _(f"The {output.serial} stock item was not passed on all of the required tests"))
```

Comment:
The build output may not necessarily have a serial number. I'd suggest that you change this to:

```python
raise ValidationError(_("This output has not passed all required tests"))
```